hotel/models/account_move.py

```diff
@@ -1,18 +1,29 @@
-# See LICENSE file for full copyright and licensing details.
-
-from odoo import api, models
+from odoo import api, models, fields
 
 
 class AccountMove(models.Model):
 
     _inherit = "account.move"
 
+    is_folio = fields.Boolean(string="Is folio", default=False)
+    folio_id = fields.Many2one(comodel_name="hotel.folio", string="Corresponding folio")
+
     @api.model
     def create(self, vals):
-        res = super(AccountMove, self).create(vals)
         if self._context.get("folio_id"):
             folio = self.env["hotel.folio"].browse(self._context["folio_id"])
+            vals.update({"is_folio": True, "folio_id": folio.id})
+            res = super(AccountMove, self).create(vals)
             folio.write(
                 {"hotel_invoice_id": res.id, "invoice_status": "invoiced"}
             )
-        return res
+            return res
```

Comment:
super call and the return should be outside the if a condition, If folio_id is not in context then It will block the orm create method process.